tests/drivers/gpio/gpio_basic_api/testcase.yaml

```diff
@@ -36,20 +36,66 @@ tests:
     depends_on:
       - gpio
       - arduino_gpio
-    filter: dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+    filter: >
+      dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+      and not CONFIG_SOC_FAMILY_STM32
     extra_args: "DTC_OVERLAY_FILE=arduino.overlay"
+    extra_configs:
+      - CONFIG_ONE_PORT=n
     integration_platforms:
       - frdm_k64f
     platform_exclude:
       # below boards are mikro bus not arduino
       - da14695_dk_usb
+      # below boards are customized
       - mimxrt595_evk/mimxrt595s/cm33
-
+      - nucleo_wb05kz
+      - nucleo_wb09ke
+      - nucleo_wba55cg
+      - nucleo_l031k6
   drivers.gpio.2pin_arduino_customized:
     min_flash: 34
     depends_on:
       - gpio
       - arduino_gpio
     filter: dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+    extra_configs:
+      - CONFIG_ONE_PORT=n
     platform_allow:
       - mimxrt595_evk/mimxrt595s/cm33
+  drivers.gpio.st_2pin_arduino:
+    min_flash: 34
+    depends_on:
+      - gpio
+      - arduino_gpio
+    filter: >
+      dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+      and dt_res_prop_addr_cmp("in-gpios","out-gpios", 0) and CONFIG_SOC_FAMILY_STM32
+    extra_args: "DTC_OVERLAY_FILE=st_arduino.overlay"
+    extra_configs:
+      - CONFIG_ONE_PORT=y
+    integration_platforms:
+      - stm32mp157c_dk2
```

Comment:
Any other one would be better (say disco_l475_iot1)

Reply:
ok

Reply:
Thanks